lib/Model/events.js

```diff
@@ -104,15 +112,27 @@ Model.prototype.emit = function(type) {
 };
 
 Model.prototype._on = EventEmitter.prototype.on;
+/**
+ * @param {string} type
+ * @param {string} pattern
+ * @param {Function} cb
+ * @param {ModelOnOptions} [options]
+ */
 Model.prototype.addListener =
-Model.prototype.on = function(type, pattern, cb) {
-  var listener = eventListener(this, pattern, cb);
+Model.prototype.on = function(type, pattern, cb, options) {
```

Comment:
It is more common in the current Racer API to have the optional `options` argument before the required callback. For example, this is the case in `model.start()`. Is that still possible with TypeScript? If so, I'd rather be consistent as much as possible